server/src/main/java/org/opensearch/index/remote/RemoteRefreshSegmentTracker.java

```diff
@@ -162,18 +207,37 @@ public class RemoteRefreshSegmentTracker {
     private final Object uploadBytesPerSecMutex = new Object();
 
     /**
-     * Provides moving average over the last N overall upload time (in nanos) as part of remote refresh.N is window size.
+     * Provides moving average over the last N upload speed (in bytes/s) of segment files downloaded from the remote store.
+     * N is window size. Wrapped with {@code AtomicReference} for dynamic changes in window size.
+     */
+    private final AtomicReference<MovingAverage> downloadBytesPerSecMovingAverageReference;
+
+    private final Object downloadBytesPerSecMutex = new Object();
```

Comment:
we would not be using this as we dont want the settings yet.

Reply:
Lets clean up this class in accordance with the same.

Reply:
Will need this to update the `downloadBytesPerSecond` moving average calculation. Will be removing the redundant code in this section to update the moving avg windows.